smartsim/_core/utils/serialize.py

```diff
@@ -0,0 +1,221 @@
+    from smartsim.entity import DBNode, Ensemble, Model
+    from smartsim.entity.dbobject import DBModel, DBScript
+    from smartsim.settings.base import BatchSettings, RunSettings
+
+
+# Many of the required fields for serialization require attrs that protected.
+# I do not want to accidentally expose something to users that should not see
+# for a prototype feature. This suppression should probably be removed before
+# this is officially released!
+#
+# pylint: disable=protected-access
+
+
+TStepLaunchMetaData = t.Tuple[
+    t.Optional[str], t.Optional[str], t.Optional[bool], str, str
+]
+
+
+def save_launch_manifest(manifest: _Manifest[TStepLaunchMetaData]) -> None:
+    manifest_dir = Path(manifest.metadata.exp_path) / ".smartsim/manifest"
+    manifest_dir.mkdir(parents=True, exist_ok=True)
+    manifest_file = manifest_dir / "manifest.json"
+
+    run_id = str(uuid.uuid4())
+    telemetry_data_root = manifest_dir / f"{manifest.metadata.exp_name}/{run_id}"
+
+    new_run = {
+        "run_id": run_id,
+        "timestamp": int(time.time_ns()),
+        "model": [
+            _dictify_model(
+                model,
+                *telemetry_metadata,
+                telemetry_data_root / "model",
+            )
+            for model, telemetry_metadata in manifest.models
+        ],
+        "orchestrator": [
+            _dictify_db(
+                db, nodes_info, telemetry_data_root / "database"
+            )
+            for db, nodes_info in manifest.databases
+        ],
+        "ensemble": [
+            _dictify_ensemble(
+                ens, member_info, telemetry_data_root / "ensemble"
+            )
+            for ens, member_info in manifest.ensembles
+        ],
+    }
+    try:
+        with open(manifest_file, "r", encoding="utf-8") as file:
+            manifest_dict = json.load(file)
+    except (FileNotFoundError, json.JSONDecodeError):
+        manifest_dict = {
+            "experiment": {
+                "name": manifest.metadata.exp_name,
+                "path": manifest.metadata.exp_path,
+                "launcher": manifest.metadata.launcher_name,
+            },
+            "runs": [new_run],
+        }
+    else:
+        manifest_dict["runs"].append(new_run)
+    finally:
+        with open(manifest_file, "w", encoding="utf-8") as file:
+            json.dump(manifest_dict, file, indent=2)
+
+
+def _dictify_model(
+    model: Model,
+    step_id: t.Optional[str],
+    task_id: t.Optional[str],
+    managed: t.Optional[bool],
+    out_file: str,
+    err_file: str,
+    telemetry_data_path: Path,
+) -> t.Dict[str, t.Any]:
+    colo_settings = (model.run_settings.colocated_db_settings or {}).copy()
+    db_scripts = t.cast("t.List[DBScript]", colo_settings.pop("db_scripts", []))
+    db_models = t.cast("t.List[DBModel]", colo_settings.pop("db_models", []))
+    return {
+        "name": model.name,
+        "path": model.path,
+        "run_settings": _dictify_run_settings(model.run_settings),
```

Comment:
Currently we have `exe_args` in the `applications` section of the dashboard, rather than have it inside `run_settings`.  I think that's something we should get aligned on.

Reply:
I moved them back to the `"models"`/application section, but I do think they should be moved back to run settings as that is where they exist in memory and (at least imo) the manifest should should follow that structure

Adding a TODO comment and created a ticket to track the work!